pyluna-pathology/luna/pathology/cli/slide_etl.py

```diff
@@ -0,0 +1,188 @@
+@click.option('-c', '--comment', required=False,
+              help='description/comments on the dataset (wrap in quotes)')
+@click.option('-dl', '--debug-limit', required=False, default=-1,
+              help='limit number of slides process, for debugging, dry_run is automatically enabled')
+@click.option('-nc', '--num_cores', required=False,
+              help='Number of cores to use', default=4)
+@click.option('--dry-run', is_flag=True,
+              help="Only print data, no data is copied or generated on disk.")
+@click.option('-m', '--method_param_path', required=False,
+              help='path to a metadata json/yaml file with method parameters to reproduce results')
+def cli(**cli_kwargs):
+    """ Ingest slides by adding them to a file or s3 based storage location and generating metadata about them
+
+    \b
+    Inputs:
+        input_slide_folder: a slide image folder to walk over to find slide images
+    \b
+    Outputs:
+        slide_table: a dataset representing the slide ingestion job
+    \b
+    Example:
+        slide_etl /source/pathology/images \
+            --project_name PATH-123-45 \
+            --comment "H&E needle core biospies" \
+            --store_url s3://localhost:9000/ \
+            --num_cores 8 \
+            -o /data/PATH-123-45/table
+    """
+    cli_runner( cli_kwargs, _params_, slide_etl)
+
+
+from pathlib import Path
+import pandas as pd
+from tqdm import tqdm
+
+from dask.distributed import Client, as_completed
+from luna.common.adapters import IOAdapter
+from luna.common.utils import rebase_schema_numeric
+
+
+import openslide
+from luna.common.utils import generate_uuid
+import numpy as np
+
+from datetime import datetime
+
+def slide_etl(input_slide_folder, project_name, num_cores, dry_run, debug_limit, store_url, comment, output_dir):
+    """ CLI tool method
+
+    Args:
+        input_slide_folder (str): path to parent directory containing slide images
+        project_name (str): project name underwhich the slides should reside
+        comment (str): comment and description of dataset
+        num_cores (int): number of cores (interfaced to dask)
+        debug_limit (int): cap number fo slides to process below this number (for debugging, testing, dry_run is automatically enabled)
+        dry_run (bool): don't actually generate or copy any data on disk, print outputs
+        output_dir (str): path to output table
+
+
+    Returns:
+        dict: metadata about function call
+    """
+    slide_paths = []
+
+    for path, _, files in os.walk(input_slide_folder):
+        for file in files:
+            file = os.path.join(path, file)
+
+            if not Path(file).suffix in VALID_SLIDE_EXTENSIONS: continue
+
+            slide_paths.append(file)
+
+
+    if debug_limit > 0: 
+        slide_paths = slide_paths[:debug_limit]
+        dry_run = True
+
+    if dry_run: logger.info ("Note, this is a dry run!!!")
+
+    logger.info(slide_paths)
```

Comment:
It would be great to optionally pass in a csv/txt file to subset what we ETL from the slide_paths.